### PR TITLE
feat: [FIR-979] OCPP structure changes

### PR DESCRIPTION
**Pull request by rob179994**
Adds rawJson to OCPP request object and make every request extend this base class instead of implementing Request directly.

Note - this branch is going to have to be force pushed into master, which when it's ready will have to be force pushed over migration branch which is what all of stuff builds from

[Related to this jira ticket](https://ohmenergy.atlassian.net/browse/FIR-979)

Related PRs:
- [OCPP lib changes](https://github.com/OhmEnergy/Java-OCA-OCPP/pull/11)
- [server-api](https://github.com/OhmEnergy/ohme-server-api/pull/2778)

### CODE REVIEW

File: ocpp-v1_6/src/main/java/eu/chargetime/ocpp/model/smartcharging/ClearChargingProfileRequest.java [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
-import eu.chargetime.ocpp.utilities.MoreObjects;
 import java.util.Objects;
+
 import javax.xml.bind.annotation.XmlElement;
 import javax.xml.bind.annotation.XmlRootElement;
 
+import eu.chargetime.ocpp.PropertyConstraintException;
+import eu.chargetime.ocpp.model.OcppRequest;
+import eu.chargetime.ocpp.model.core.ChargingProfilePurposeType;
+import eu.chargetime.ocpp.utilities.MoreObjects;
```
Comment (DanielVocke) [inline ocpp-v1_6/src/main/java/eu/chargetime/ocpp/model/smartcharging/ClearChargingProfileRequest.java:6]: I assume all those changes happened through auto import. Means lots of unnecessary changes in the files (that will likely cause merge conflicts once we pull in from the fork). Not a show stopper, but next time try to minimize by not applying auto format / import.

**Review by DanielVocke — APPROVED** [review]

**Comment by DanielVocke** [comment]
where are those conflicts from? Are you merging into the correct branch?

File: ocpp-v1_6/src/main/java/eu/chargetime/ocpp/model/smartcharging/ClearChargingProfileRequest.java (hunk 1 above)
Comment (rob179994) [reply ocpp-v1_6/src/main/java/eu/chargetime/ocpp/model/smartcharging/ClearChargingProfileRequest.java:6]: Removed and minimised :)

File: ocpp-common/src/main/java/eu/chargetime/ocpp/model/OcppRequest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	@Override
+	public String getRawMessageJson(){
+		return rawMessageJson;
+	}
+
+	@Override
+	public String getRequestId(){
+		// id needs to be in the 2nd position of the array after a first item that's not a string.
+		// this split logic won't work if the first item is also a string
+		return rawMessageJson.split("\"")[1];
```
Comment (PhilPotter) [inline ocpp-common/src/main/java/eu/chargetime/ocpp/model/OcppRequest.java:24]: Maybe just mention this to FW team as an informational thing.

**Review by PhilPotter — APPROVED** [review]
LGTM.
